### PR TITLE
cool whip and baton changes

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
Whips and telescopic/police batons now have more tactical limb-based combat applications, no longer being as much crutch weapons as they were, but still being robust
TL;DR: whips now feel like actual Indiana Jones shit, telebatons now require a bit of tactical thinking and robustness, whilst not being neutered

Whips:
have 3 range
no longer deal damage
when targeting an arm, knock the item out of that hand. If your other hand is empty, you will automatically catch the item and switch hands to it
image
do knockdown when targeting the legs, unless you are adjacent to the target 
do a half-second truestun (without causing item drops) when targeting the chest, head, or groin, and cause the target to scream, unless the target is wearing somewhat decent melee armor

Telebatons:
cooldown is halved
now deal stamina damage when targeting head or torso. now affected by armor.
deal knockdown when targeting legs
when targeting an arm, instantly disables it, depending on target armor. internally, this is 50 stamina damage to the arm, affected by armor.

NOTE: needs https://github.com/BeeStation/BeeStation-Hornet/pull/1292 (knockdowns and truestuns no longer drop items by default). Otherwise, this isn't balanced

## Why It's Good For The Game

telebatons and the whip are less in the vein of fightending unga weapons and now have a degree of tactics and precisions required, but are still quite robust. 

## Changelog
:cl:
tweak: whips now feel like indiana jones and shit man. 
balance: telebatons need a bit more robustness
/:cl:

### CODE REVIEW

**Review by Crossedfall — DISMISSED** [review]
Rebase and remove the old commits.

**Comment by zeskorion** [comment]
> Rebase and remove the old commits.

done

**Comment by qwertyquerty** [comment]
I dunno about the whip steal items thing. Seems like a really easy "haha mine now"

**Comment by zeskorion** [comment]
> I dunno about the whip steal items thing. Seems like a really easy "haha mine now"

Whips already call dropallhelditems on hit, AND do fifteen damage at the moment. Making it need deliberate arm targeting is actually something of a nerf, at least in straight up combat potential

File: code/game/objects/items/melee/misc.dm
Comment (qwertyquerty) [inline code/game/objects/items/melee/misc.dm:520]: You can probably combine the two parts of this into one here.

Comment (zeskorion) [reply code/game/objects/items/melee/misc.dm:520]: due to it targeting the specific arm's item, it'd be a bit janky as i'd code it, so this is just easier to do.

**Comment by zeskorion** [comment]
hold off on the merge for this. something broke with https://github.com/BeeStation/BeeStation-Hornet/pull/1365 so it no longer works on truestuns

**Comment by zeskorion** [comment]
removed the stun for now until i fix #1365, this is now good to merge

**Comment by zeskorion** [comment]
this bullshit with spiked skin is pissing me off

**Comment by zeskorion** [comment]
this shit is up to date with master, no idea what's going on with spiked skin, lemme check it out

**Pull request merged**
